bootloader/grub.go

```diff
@@ -483,3 +483,59 @@ func (g *grub) TrustedAssets() ([]string, error) {
 		"EFI/boot/grubx64.efi",
 	}, nil
 }
+
+// RecoveryBootChain returns the chain of recovery mode boot files for the
+// specified recovery system and kernel. This function should be called
+// only for the recovery bootloader.
+func (g *grub) RecoveryBootChain(kernelPath string) ([]BootFile, error) {
+	if !g.recovery {
+		return nil, fmt.Errorf("not a recovery bootloader")
+	}
+
+	recoveryTrustedAssets, err := g.TrustedAssets()
+	if err != nil {
+		return nil, err
+	}
+
+	// add trusted assets to the recovery chain
+	chain := make([]BootFile, 0, len(recoveryTrustedAssets)+1)
+	for _, ta := range recoveryTrustedAssets {
+		chain = append(chain, NewBootFile("", ta, RoleRecovery))
+	}
+	// add recovery kernel to the recovery chain
+	chain = append(chain, NewBootFile(kernelPath, "kernel.efi", RoleRecovery))
+
+	return chain, nil
+}
+
+// BootChain returns the chain of run mode boot files for the specified run
+// mode bootloader and kernel. This function should be called only for the
+// recovery bootloader.
```

Comment:
Maybe append something like: "only for the recover bootloader because run mode always chain loads the recover bootloader first" or something along these lines?

Reply:
We could obtain the run mode boot chain from the run mode bootloader if we pass the recovery bootloader as a parameter instead. It was our decision to query the recovery bootloader for this piece of information, we just need both bootloaders to compose the chain.

Reply:
same consideration about the comment.